libraries/bsg_manycore.cpp

```diff
@@ -708,24 +714,15 @@ static int hb_mc_manycore_send_read_rqst(hb_mc_manycore_t *mc,
 
         // mark request with id
         hb_mc_request_packet_set_load_id(&rqst.request, id);
-        int shift = hb_mc_npa_get_epa(npa) & 0x3;
-        /* set the byte mask */
-        switch (sz) {
-        case 4:
-                hb_mc_request_packet_set_mask(&rqst.request, HB_MC_PACKET_REQUEST_MASK_WORD);
-                break;
-        case 2:
-                hb_mc_request_packet_set_mask(&rqst.request,
-                                              static_cast<hb_mc_packet_mask_t>(
-                                                      HB_MC_PACKET_REQUEST_MASK_SHORT << shift));
-                break;
-        case 1:
-                hb_mc_request_packet_set_mask(&rqst.request, static_cast<hb_mc_packet_mask_t>(
-                                                      HB_MC_PACKET_REQUEST_MASK_BYTE << shift));
-                break;
-        default:
-                return HB_MC_INVALID;
-        }
+
+        // set load info
+        hb_mc_request_packet_load_info_t info = {};
+        info.part_sel       = hb_mc_npa_get_epa(npa) & 0x3;
+        info.is_unsigned_op = 1;
```

Comment:
Hmmmm. I think we did this before, but don't we have a templated read function somewhere?

Reply:
Plus a test to verify these work

Reply:
It calls this.

Reply:
Then we should have some way of setting is_unsigned. Or file an issue for it.

Reply:
When do we do a signed read? is_unsigned is not is_word. It's a flag to indicate if the returned data should be sign extended.

Reply:
AFAIK our model in CUDA-Lite is to read/write data 'as is' and support for sign extension in our API would be a new feature.

Reply:
Worth filing an issue - no urgency to fix

Reply:
I don't even think its worth that. This would be a feature nobody's asking for.

Reply:
See test_vcache_flush for a test that exercises this change.

Reply:
Wait, did you implement it?